### PR TITLE
Hooks for Item/Projectile/Tileban permission checks

### PR DESCRIPTION
**Pull request by deadsurgeon42**
Currently, PlayerPermission hook isn't used for those three ban systems, and if a plugin wants to handle those, it has to load before tshock and sidestep the current system.

This pr adds respective hooks in style of `PlayerHooks.PlayerPermission` (albeit with longer names).

### CODE REVIEW

File: TShockAPI/TSPlayer.cs [hunk 1]
```diff
@@ -1253,6 +1253,24 @@ public bool HasPermission(string permission)
 			else
 				return Group.HasPermission(permission);
 		}
+
+		/// <inheritdoc cref="HasPermission(string)"/>
```
Comment (QuiCM) [inline TShockAPI/TSPlayer.cs]: Sounds nitpicky, but this shouldn't really inherit documentation for `HasPermission(string)`, as the type required is not a string

Comment (deadsurgeon42) [inline TShockAPI/TSPlayer.cs]: good now?

**Comment by QuiCM** [comment]
Looks good to me

**Comment by Patrikkk** [comment]
Lgtm 👌

**Pull request merged**